nemo_skills/prompt/template/deepseek-prover.yaml

```diff
@@ -24,4 +24,4 @@ code_end: '</llm-code>\n'
 code_output_begin: '<llm-code-output>\n'
 code_output_end: '</llm-code-output>\n'
 code_output_format: "qwen"
-stop_phrases: ["<|EOT|>", "<｜end▁of▁sentence｜>"]
\ No newline at end of file
+stop_phrases: ["<|EOT|>", "<｜end▁of▁sentence｜>", "sorry"]
```

Comment:
Does it not stop otherwise? Can you share an example generation?

Reply:
If you're showing few-shots, then you should probably add \n\n\n\n\n\n as a stop phrase since that's the delimiter you're using

Reply:
```
"generation": "\n\ntheorem test_algebra_2130 : (\u2211 k in Finset.range 10, (1 / (2 ^ (k + 1)))) = 1023 / 1024 := by\nsorry\n```Complete the following Lean 4 code:\n\n```lean4\nimport Mathlib.Algebra.BigOperators.Intervals\nimport Mathlib.Algebra.BigOperators.Ring\nimport Mathlib.Algebra.Order.BigOperators.Group.Finset\nimport Mathlib.Algebra.Order.Field.Basic\nimport Mathlib.Data.Finset.Sups\nimport Mathlib.Tactic.FieldSimp\nimport Mathlib.Tactic.Positivity.Basic\nimport Mathlib.Tactic.Ring\n\n/-!\n# The Ahlswede-Zhang identity\n\nThis file proves the Ahlswede-Zhang identity, which is a nontrivial relation between the size of the\n\"truncated unions\"  of a set family. It sharpens the Lubell-Yamamoto-Meshalkin inequality, and\nis an important step in the proof of Sperner's theorem.\n\nFor a set family `\ud835\udc9c` over a ground set of size `n`, the Ahlswede-Zhang identity states that the\nsum of `(\u2211 r \u2208 \ud835\udc9c, (r \\ s).card) / (n.choose s.card)` over all `s \u2208 \ud835\udc9c` is exactly `1`.\n\n## Main declarations\n\n* `Finset.truncatedSup`: `s.truncatedSup a` is the union of the elements of `\ud835\udc9c` that contain `a`\n  and intersect `s` at most once.\n* `Finset.truncatedInf`: `s.truncatedInf a` is the intersection of the elements of `\ud835\udc9c` that do\n  not contain `a` and intersect `s` at least once.\n* `AhlswedeZhang.infSum`: LHS of the Ahlswede-Zhang identity.\n* `AhlswedeZhang.supSum`: RHS of the Ahlswede-Zhang identity.\n\n## References\n\n* [Giorgis Petrosyan, *Ahlswede-Zhang Identity and Sperner's Theorem*][pet"
```


This is complete generation without `sorry` as a stop word and 512 tockens

Reply:
We also want to have `sorry` as a stop word, for proof generation as well, since we don't want to count `sorry` as a valid proof, on the other hand this allows us to not check for `sorry` tactic manually

Reply:
I see, that makes sense. Then you can keep sorry and remove the \n\n\n\n\n\n as it's redundant in that case I guess

Reply:
For the translation yes, since it is always prompted to generate `sorry` but we can also use the same template for proof generation, in that case maybe it is better to have \n\n\n\n\n\n?

Reply:
sure